lib/subscription/subscription-buffer.ts

```diff
@@ -0,0 +1,63 @@
+/// <reference path='../../typings/tsd.d.ts' />
+
+import Subscription = require('./subscription');
+import Buffer = require('./buffer');
+
+export = SubscriptionWithBuffer;
+
+class SubscriptionWithBuffer extends Subscription {
+
+    private buffer: Buffer<Object>;
+
+    constructor( cid: number,
+                 security: string,
+                 fields: string[],
+                 maxBufferLength: number,
+                 options?: any) {
+        super(cid, security, fields, options);
+        this.buffer = new Buffer<Object>(maxBufferLength);
+    }
+
+    pushBuffer(data: any): void {
+        this.buffer.push(data);
+    }
+
+    getNewBuffer(): Object {
+        var buff = this.buffer.getNew();
+        return {
+            'correlationId': this.correlationId,
+            'data': buff[0],
+            'missed_ticks': buff[1]
+        };
+    }
+
+    getOldBuffer(): Object {
+        var buff = this.buffer.getOld();
+        return {
+            'correlationId': this.correlationId,
+            'data': buff[0],
```

Comment:
I'm not sure I understand how buff[0] and buff[1] map to the data/missed_ticks; getOld (and getNew, for the previous function) return an array of multiple values so you should just return the entire array?